citadel/indico_citadel/util.py

```diff
@@ -67,15 +67,20 @@ def format_query(query, placeholders):
     :param placeholders: placeholder whitelist
     :returns escaped query
     """
-    patt = r'({}):([^:"\s]+|"[^"]+")\s*'.format('|'.join(placeholders.keys()))
-    # Extract all placeholders
-    p = [f'+{placeholders[x.group(1)]}:{x.group(2)}'
-         for x in re.finditer(patt, query) if x.group(1) in placeholders]
-    # Escape keyword based arguments
-    query = escape(re.sub(patt, '', query)).strip()
-    if query:
-        p.append(f'+({query})')
-    return ' '.join(p)
+    patt = r'(?:^|\s)({}):([^:"\s]+|"[^"]+")(?:$|\s)'.format('|'.join(placeholders.keys()))
```

Comment:
```suggestion
    patt = r'(?:^|\s)({}):([^:"\s]+|"[^"]+")(?:$|\s)'.format('|'.join(map(re.escape, placeholders)))
```

`.keys()` isn't needed and while i would never expect any regex-special chars in a placeholder it feels cleaner to run them through `re.escape()` anyway